dashboard/modules/stats_collector/tests/test_stats_collector.py

```diff
@@ -417,5 +420,82 @@ class InfeasibleActor:
                 raise Exception(f"Timed out while testing, {ex_stack}")
 
 
+def test_actor_pubsub(disable_aiohttp_cache, ray_start_with_dashboard):
+    timeout = 5
+    assert (wait_until_server_available(ray_start_with_dashboard["webui_url"])
+            is True)
+    address_info = ray_start_with_dashboard
+    address = address_info["redis_address"]
+    address = address.split(":")
+    assert len(address) == 2
+
+    client = redis.StrictRedis(
+        host=address[0],
+        port=int(address[1]),
+        password=ray_constants.REDIS_DEFAULT_PASSWORD)
+
+    p = client.pubsub(ignore_subscribe_messages=True)
+    p.psubscribe(ray.gcs_utils.RAY_ACTOR_PUBSUB_PATTERN)
+
+    @ray.remote
+    class DummyActor:
+        def __init__(self):
+            pass
+
+    # Create a dummy actor.
+    a = DummyActor.remote()
+
+    def handle_pub_messages(client, msgs, timeout, expect_num):
+        start_time = time.time()
+        while time.time() - start_time < timeout and len(msgs) < expect_num:
+            msg = client.get_message()
+            if msg is None:
+                time.sleep(0.01)
+                continue
+            pubsub_msg = ray.gcs_utils.PubSubMessage.FromString(msg["data"])
+            actor_data = ray.gcs_utils.ActorTableData.FromString(
+                pubsub_msg.data)
+            msgs.append(actor_data)
+
+    msgs = []
+    handle_pub_messages(p, msgs, timeout, 2)
+
+    # Assert we received published actor messages with state DEPENDENCIES_UNREADY and ALIVE.
+    assert len(msgs) == 2
+
+    # Kill actor.
+    ray.kill(a)
+    handle_pub_messages(p, msgs, timeout, 3)
+
+    # Assert we received published actor messages with state DEAD.
+    assert len(msgs) == 3
+
+    def actor_table_data_to_dict(message):
+        return dashboard_utils.message_to_dict(
+            message, {
+                "actorId", "parentId", "jobId", "workerId", "rayletId",
+                "actorCreationDummyObjectId", "callerId", "taskId",
+                "parentTaskId", "sourceActorId", "placementGroupId"
+            },
+            including_default_value_fields=False)
+
+    for msg in msgs:
+        actor_data_dict = actor_table_data_to_dict(msg)
+        # DEPENDENCIES_UNREADY is 0, which would not be keeped in dict. We need check its original value.
+        if msg.state == 0:
+            assert len(actor_data_dict) > 5
+            keys = ("actorId", "jobId", "taskSpec")
+            for k in keys:
+                assert k in actor_data_dict
+        # For status that is not DEPENDENCIES_UNREADY, only states fields will be published.
+        elif actor_data_dict["state"] in ("ALIVE", "DEAD"):
+            assert len(actor_data_dict) == 4
+            keys = ("state", "address", "timestamp", "pid")
```

Comment:
Maybe `assert actor_data_dict.keys() == {"state", "address", "timestamp", "pid"}` is more concise?

From the Python doc: https://docs.python.org/3/library/stdtypes.html#dictionary-view-objects
> Keys views are set-like since their entries are unique and hashable. For set-like views, all of the operations defined for the abstract base class collections.abc.Set are available (for example, ==, <, or ^).